src/Microsoft.AspNetCore.Mvc.Razor.Host/BaseDocumentClassifierPass.cs

```diff
@@ -0,0 +1,177 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Globalization;
+using System.IO;
+using Microsoft.AspNetCore.Razor.Evolution;
+using Microsoft.AspNetCore.Razor.Evolution.Intermediate;
+using Microsoft.Extensions.Primitives;
+
+namespace Microsoft.AspNetCore.Mvc.Razor.Host
+{
+    public abstract class BaseDocumentClassifierPass : IRazorIRPass
```

Comment:
We should put this in Razor. Most document classifiers we'd write will do 80% of the stuff the same.